client/src/components/UploadForm.vue

```diff
@@ -171,142 +203,159 @@
+      ),
+      report: flightReport.value,
+    });
+    if (response.status != 200) throw response.statusText;
 
-      var reader = new FileReader();
-      reader.onload = function () {
-        var output = document.getElementById("preview-image");
-        output.src = reader.result;
-      };
-      reader.readAsDataURL(event.target.files[0]);
-      this.imageUploadButtonDisabled = false;
-    },
-    async uploadImages() {
-      // TODO: Handle multiple file upload
-      try {
-        const formData = new FormData();
-        formData.append("image", this.userImages[0], this.userImages[0].name);
-        formData.append("flightId", this.flightId);
-        formData.append("userId", getUserId);
-        const response = await ApiService.uploadImages(formData);
-        if (response.status != 200) throw response.statusText;
-        this.imageUploadSuccessfull = true;
-        this.imageUploadButtonDisabled = true;
-      } catch (error) {
-        console.log(error);
-      }
-    },
-    redirectToFlight(id) {
-      this.$router.push({
-        name: "Flight",
-        params: {
-          flightId: id,
-        },
-      });
+    uploadedPhotos.value.forEach((e) => addPhotoDescription(e.id, e));
+
+    redirectToFlight(externalId.value);
+  } catch (error) {
+    console.log(error);
+  }
+};
+
+// Photos
+
+const addPhotoButtonIsEnabled = computed(() => flightId.value != null);
+
+const photoInput = ref(null);
+onMounted(() => {
+  photoInput.value = document.getElementById("photo-input");
+});
+
+const selectedPhotos = ref([]);
+const uploadedPhotos = ref([]);
+
+const onAddPhoto = () => photoInput.value.click();
+
+const onDeletePhoto = (id) => {
+  const index = uploadedPhotos.value.findIndex((e) => e.id === id);
+  if (index > -1) uploadedPhotos.value.splice(index, 1);
+};
+
+const onPhotoSelected = (event) => {
+  [...event.target.files].forEach((element) => {
+    selectedPhotos.value.push(element);
+    uploadPhoto(element);
+  });
+};
+
+const addPhotoDescription = async (id, description) => {
+  await ApiService.editPhoto(id, description);
+};
+
+const uploadPhoto = async (photo) => {
+  const index = uploadedPhotos.value.length;
+  const _phototmp = { id: null, description: "" };
+  uploadedPhotos.value[index] = _phototmp;
+
+  try {
+    const formData = new FormData();
+    formData.append("image", photo, photo.name);
+    formData.append("flightId", flightId.value);
+    formData.append("userId", getUserId);
+    console.log(photo.name, flightId.value);
+    const res = await ApiService.uploadPhotos(formData);
+    if (res.status != 200) throw res.statusText;
+    uploadedPhotos.value[index].id = res.data.id;
+  } catch (error) {
+    console.log(error);
+    uploadedPhotos.value[index].id = "failed";
+  }
+};
+const redirectToFlight = (id) => {
+  router.push({
+    name: "Flight",
+    params: {
+      flightId: id,
     },
-  },
+  });
 };
 </script>
 
 <style scoped>
-.container {
```

Comment:
Ich kann mir vorstellen warum du das hier entfernt hast. Aber vorher hat es mir optisch besser gefallen.

Vielleicht sollten wir das hier Analog zu UserRegister, UserActivate machen und das in eine Card packen?

Reply:
gleicher gedanke